zerver/lib/onboarding.py

```diff
@@ -191,7 +187,23 @@ def send_welcome_bot_response(send_request: SendMessageRequest) -> None:
     to welcome-bot, trigger the welcome-bot reply."""
     welcome_bot = get_system_bot(settings.WELCOME_BOT, send_request.realm.id)
     human_response_lower = send_request.message.content.lower()
+    human_user_id = send_request.message.sender.id
     content = select_welcome_bot_response(human_response_lower)
+    realm_id = send_request.realm.id
+
+    if (
+        Message.objects.filter(
+            realm_id=realm_id,
+            sender_id=welcome_bot.id,
+            recipient__type_id=human_user_id,
+            content__icontains=bot_commands(),
+        ).exists()
```

Comment:
This is the query plan:

```
zulip=> explain analyze SELECT 1 AS "a" FROM "zerver_message" INNER JOIN "zerver_recipient" ON ("zerver_message"."recipient_id" = "zerver_recipient"."id") WHERE (UPPER("zerver_message"."content"::text) LIKE UPPER('%`apps`, `profile`, `theme`, `channels`, `topics`, `message formatting`, `keyboard shortcuts`, `help`.%') AND "zerver_message"."realm_id" = 2 AND "zerver_recipient"."type_id" = 10 AND "zerver_message"."sender_id" = 5) LIMIT 1;
                                                                            QUERY PLAN                                                                            
------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=0.14..18.00 rows=1 width=4) (actual time=0.031..0.031 rows=0 loops=1)
   ->  Nested Loop  (cost=0.14..18.00 rows=1 width=4) (actual time=0.030..0.030 rows=0 loops=1)
         ->  Seq Scan on zerver_recipient  (cost=0.00..1.64 rows=2 width=4) (actual time=0.006..0.010 rows=2 loops=1)
               Filter: (type_id = 10)
               Rows Removed by Filter: 78
         ->  Index Scan using zerver_message_realm_sender_recipient on zerver_message  (cost=0.14..8.17 rows=1 width=4) (actual time=0.009..0.009 rows=0 loops=2)
               Index Cond: ((realm_id = 2) AND (sender_id = 5) AND (recipient_id = zerver_recipient.id))
               Filter: (upper(content) ~~ '%`APPS`, `PROFILE`, `THEME`, `CHANNELS`, `TOPICS`, `MESSAGE FORMATTING`, `KEYBOARD SHORTCUTS`, `HELP`.%'::text)
 Planning Time: 23.432 ms
 Execution Time: 0.081 ms
(10 rows)

```

@timabbott do you think sequential scan on Recipient table is fine ? -- I expect the table to be huge...

Reply:
I dont have any Idea how to do this can you please refer me to some docs, Thanks again

Reply:
See: https://github.com/zulip/zulip/pull/30278#discussion_r1659013545

Reply:
Thank you so much